### PR TITLE
Inline link card: Style updates to better match original design

### PR DESCRIPTION
**Pull request by kellychoffman**
Before: 
Screen Shot 2021-06-23 at 3 04 43 PM

After: 
Screen Shot 2021-06-23 at 2 55 48 PM

- site icon visually spans height of name+URL rows
- increase line height between name and URL, make URL larger
- increase line height of description
- align “open in new tab” toggle with the left content side.

### CODE REVIEW

File: packages/block-editor/src/components/link-control/style.scss
Comment (jasmussen) [inline packages/block-editor/src/components/link-control/style.scss]: We should try to use existing SCSS variables wherever possible. You can find a list of those in `gutenberg/packages/base-styles/_variables.scss`:

Screenshot 2021-06-24 at 14 48 16

In this case, you could use `$grid-unit-40` instead of `32px`.

There's also a handy `_colors.scss` in the same package.

File: packages/block-editor/src/components/link-control/style.scss
Comment (jasmussen) [inline packages/block-editor/src/components/link-control/style.scss]: The `$gray-0` color is only for mobile native, and causes a compile error for web:

Screenshot 2021-06-24 at 14 59 54

It's probably meant to be `$gray-100`:

Screenshot 2021-06-24 at 14 51 38

The fact that we have separate colors for mobile native, I'm not sure is a good idea. But separate to this PR.

File: packages/block-editor/src/components/link-control/style.scss
Comment (jasmussen) [inline packages/block-editor/src/components/link-control/style.scss]: In general we don't use em for font sizes or lineheights unless in edgecases. I'm aware this was existing code and you're moving it around, but since you're in here it might be nice to update. 1.1em here computes to 14.3px font size and the line height and margins aren't that round either:

Screenshot 2021-06-24 at 15 06 35

If I remove the top margin, change the font size to `$default-font-size`, and set the line-height of the parent container to `1.5` (unitless), I get this:

Screenshot 2021-06-24 at 15 08 31

Happy to push these changes, by the way, just let me know.

File: packages/block-editor/src/components/link-control/style.scss
Comment (jasmussen) [inline packages/block-editor/src/components/link-control/style.scss]: Same here with ems, this computes to a non even number:

Screenshot 2021-06-24 at 15 10 07

I'd just use $default-font-size here as well, for legibility reasons.

File: packages/block-editor/src/components/link-control/style.scss
Comment (jasmussen) [inline packages/block-editor/src/components/link-control/style.scss]: I'd still use the grid units here. You can replace the 16px with $grid-unit-20 plainly.

**Review by jasmussen — COMMENTED** [review]
Nice work. There's one color that has to be fixed before this can land as it won't compile otherwise. 

Before:

before

After:

after


The rest of my comments are in the nitpicks category, but I think they're worth addressing while you're here. I'm also happy to push these changes, just let me know.

**Comment by jasmussen** [comment]
Unrelated to this PR, the description uses the `Text` component which still outputs `color: #000000;` regardless of where it's used. We should remove that definition at the source:

Screenshot 2021-06-24 at 15 03 50

**Review by getdave — COMMENTED** [review]
Thanks for the improvements! @jasmussen's comments cover all the things I might have said so consider this my 👍 .

File: packages/block-editor/src/components/link-control/style.scss
Comment (kellychoffman) [reply packages/block-editor/src/components/link-control/style.scss]: I had checked out the colors, but failed to notice the grid units. Thanks for the tip!!

**Comment by jasmussen** [comment]
I pushed some polish to variable-ize things. Things should look virtually the same:

Screenshot 2021-06-25 at 16 15 12

**Review by jasmussen — APPROVED** [review]

**Comment by getdave** [comment]
Looking forward to seeing this merged 👍👍

**Comment by Mamaduka** [comment]
Hi, @kellychoffman

Can you rebase this PR if you have time? And we can merge it.

Thanks

**Comment by Mamaduka** [comment]
Thanks, @jasmussen :hero:

**Comment by jasmussen** [comment]
I took the liberty of rebasing this to trunk. It now looks like this:

Screenshot 2021-09-07 at 12 53 53

This is what trunk looks like:

Screenshot 2021-09-07 at 12 58 14

**However, we probably can't land this one as-is**. Because during the rebase I found two adjacent issues that need a little thought.

Some changes happened in #33652 to fix an icon size issue with link search results:

Screenshot 2021-09-07 at 12 59 25

As it turns out, the CSS class `block-editor-link-control__search-item-header` is used _both_ by this link card, but also by by the link search interface. Because this same class is used in both places, link search now has a gray background in this branch, a fact I missed in the first review: 

Screenshot 2021-09-07 at 13 02 36

Also, #33753 changed that same CSS class to allow word-wrapping, which if we had kept it would've looked like this:

Screenshot 2021-09-07 at 12 56 40

I don't have a strong opinion, but kept the text-eliding that this PR added. Though since it basically doubles back on the work by @getdave, I thought I'd surface it.

The most important next step, however, is to figure out how to best handle that this CSS class applies to both link search results and inside the link card.

File: packages/block-editor/src/components/link-control/style.scss
Comment (getdave) [inline packages/block-editor/src/components/link-control/style.scss:174]: I'd be concerned about changing this. The referenced issue provides a clear rationale for why this was necessary.

File: packages/block-editor/src/components/link-control/style.scss
Comment (getdave) [inline packages/block-editor/src/components/link-control/style.scss:181]: Just to be clear we changed this behaviour to allow much more text to display. This is because it better mirrors the classic editor link behaviour and also it makes it more optimised for selecting items when you need more context.

With only a small preview of text you sometimes cannot tell the different between two items. I don't think we should roll back on this behaviour. It was requested by several 3rd party contributors.

File: packages/block-editor/src/components/link-control/style.scss
Comment (getdave) [inline packages/block-editor/src/components/link-control/style.scss:236]: ❤️

**Review by getdave — CHANGES_REQUESTED** [review]
I love lots about this PR - especially all the great work that's gone into using variables and sorting out some accessibility issues with animation that I missed. That work is much appreciated.

I do however, have concerns about reverting changes to the wrapping of text. In https://github.com/WordPress/gutenberg/pull/33753 we deliberately made text wrap in order to solve the UX problem whereby users cannot tell two links apart without the additional context of being able to see the full link text.

If the changes to wrapping can be reverted to match what's on `trunk` now then we should look to merge this one.

Thanks @jasmussen for the ping 🙇

**Comment by getdave** [comment]
@kellychoffman Shall we look to land this? Lots of great changes here it would be good to have in place.

**Comment by getdave** [comment]
Closing as the UI has evolved significantly.

**Pull request closed without merging**
